packages/oauth/oauth-provider/src/token/verify-token-claims.ts

```diff
@@ -17,48 +18,50 @@ export type VerifyTokenClaimsResult = {
   token: OAuthAccessToken
   tokenId: TokenId
   tokenType: OAuthTokenType
-  claims: SignedTokenPayload
+  tokenClaims: SignedTokenPayload
+  dpopResult: null | DpopResult
 }
 
 export function verifyTokenClaims(
   token: OAuthAccessToken,
   tokenId: TokenId,
   tokenType: OAuthTokenType,
-  dpopJkt: string | null,
-  claims: SignedTokenPayload,
+  tokenClaims: SignedTokenPayload,
+  dpopResult: null | DpopResult,
   options?: VerifyTokenClaimsOptions,
 ): VerifyTokenClaimsResult {
   const dateReference = Date.now()
-  const claimsJkt = claims.cnf?.jkt ?? null
 
-  const expectedTokenType: OAuthTokenType = claimsJkt ? 'DPoP' : 'Bearer'
+  const expectedTokenType: OAuthTokenType = tokenClaims.cnf?.jkt
+    ? 'DPoP'
+    : 'Bearer'
   if (expectedTokenType !== tokenType) {
     throw new InvalidTokenError(expectedTokenType, `Invalid token type`)
   }
-  if (tokenType === 'DPoP' && !dpopJkt) {
-    throw new InvalidDpopProofError(`jkt is required for DPoP tokens`)
+  if (tokenType === 'DPoP' && !dpopResult) {
+    throw new InvalidDpopProofError(`DPoP proof required`)
   }
-  if (claimsJkt !== dpopJkt) {
+  if (tokenClaims.cnf?.jkt !== dpopResult?.jkt) {
```

Comment:
This looks a little bit like we want to permit a match when they are both missing, though I expect that isn't the case.  And the fact that these were both `null` and are now both `undefined` when missing is messing with me a little.  Would it make sense to explicitly rule out the missing cases?
```ts
if (tokenType === 'DPoP') {
  assert(tokenClaims.cnf?.jkt, 'cnf.jkt claim must exist via expected token type check')
  if (!dpopResult) {
    throw new InvalidDpopProofError(`DPoP proof required`)
  }
  if (tokenClaims.cnf.jkt !== dpopResult.jkt) {
    throw new InvalidDpopKeyBindingError()
  }
}
```